src/lib/backoff.ts

```diff
@@ -0,0 +1,101 @@
+import { strict as assert } from 'assert';
+import { promisify } from 'util';
+
+import { StatusError } from './errors';
```

Comment:
I don't think this module should depend on StatusError, since it only needs a specific error interface. 

```
// This is just because is nice to have for users of the module
export interface RetryError extends Error {
	retryAfter: number;
}

// Type guard only checks that retryAfter is present
const isRetryError = (x: unknown): x is RetryError =>
	x != null && x instanceof Error && Number.isInteger((x as RetryError).retryAfter);
```

The type guard will work with any error with the right interface, including `StatusError`

Reply:
that's a great point..going with your suggestion